### PR TITLE
Add --actually-quiet parameter to check/pytest and use it in continuous integration runs

### PR DESCRIPTION
**Pull request by Strilanc**
- This fixes an annoyance where error output is hidden waaaay down at the bottom of travis-ci logs

### CODE REVIEW

File: check/pytest [hunk 1]
```diff
@@ -4,9 +4,11 @@
 # Runs pytest on the repository.
 #
 # Usage:
-#     check/pytest [--flags] [file-paths-relative-to-repo-root]
+#     check/pytest [--actually_quiet] [--flags for pytest] [file-paths-relative-to-repo-root]
```
Comment (cduck) [inline check/pytest]: The flag should probably have a dash, not an underscore to be consistent with other flags.  `--actually-quiet`.

File: check/pytest [hunk 2]
```diff
@@ -15,4 +17,20 @@
...
+        PYTEST_ARGS+=("${arg}")
+    fi
+done
+
+if [ -z "${ACTUALLY_QUIET}" ]; then
+    pytest "${PYTEST_ARGS[@]}"
+else
+    # Filter out lines like "...F....x...      [ 42%]", with coloring.
+    pytest -q --color=yes "${PYTEST_ARGS[@]}" |
+        grep -Pv '^(.\[0m)?[\.FEsx]+(.\[36m)?\s+\[\s*\d+%\](.\[0m)?$'
```
Comment (cduck) [inline check/pytest]: This seems kind of brittle but nothing bad will happen if pytest updates and it breaks.

Comment (Strilanc) [reply check/pytest]: Yeah, I'm more worried about hiding real test output than I am about perfectly suppressing pytest's output.

Unfortunately it seems to have missed pytest's output on travis in the windows tests, so I need to tweak it at least a little.

File: check/pytest (hunk 1 above)
Comment (Strilanc) [reply check/pytest]: Done.

File: check/pytest (hunk 2 above)
Comment (Strilanc) [reply check/pytest]: I have no idea how to fix the windows thing. For now I will defer it. https://github.com/quantumlib/Cirq/issues/1826

**Pull request merged**